crates/libs/bindgen/src/types/interface.rs

```diff
@@ -461,7 +461,11 @@ impl Interface {
                         type Item = T;
 
                         fn next(&mut self) -> Option<Self::Item> {
-                            let result = self.Current().ok();
+                            let result = if self.HasCurrent().unwrap_or(false) {
+                                self.Current().ok()
+                            } else {
+                                None
+                            };
```

Comment:
This is the only change - the rest are the output of code generation used for validation.